packages/vite-plugin-cloudflare/src/index.ts

```diff
@@ -316,6 +316,26 @@ export function cloudflare(pluginConfig: PluginConfig = {}): vite.Plugin[] {
 				});
 			},
 		},
+		// Plugin to support `.wasm?init` extension
+		{
+			name: "vite-plugin:wasm-helper",
+			enforce: "pre",
+			applyToEnvironment(environment) {
+				return getWorkerConfig(environment.name) !== undefined;
+			},
+			load(id) {
+				if (!id.endsWith(".wasm?init")) {
+					return;
+				}
+
+				return `
+					import wasm from "${cleanUrl(id)}";
```

Comment:
I assume `cleanUrl()` here makes sure this isn't circular?

Reply:
It trims off the query parameter so that the import is then resolved with the `.wasm` extension.